### PR TITLE
feat(objectviewer): add openFirst, remove root label, perf improvement

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

ObjectViewer display a 'root' label which should not be required
To open the first item we have to push an initialState
Sub objects are rendered even if not opened. This push thousand of components for nothing ('hidden')

**What is the chosen solution to this problem?**

Add openFirst props
remove 'root' default label
do not render sub components if not opened

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/ObjectViewer/ObjectViewer.container.js [hunk 1]
```diff
@@ -95,6 +96,9 @@ class ObjectViewer extends React.Component {
 
 	render() {
 		const state = (this.props.state || DEFAULT_STATE).toJS();
+		if (this.props.state === DEFAULT_STATE && this.props.openFirst) {
```
Comment (ncomont) [inline packages/containers/src/ObjectViewer/ObjectViewer.container.js]: Did you mean `(state === DEFAULT_STATE && this.props.openFirst)` ?

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 2]
```diff
@@ -257,24 +257,26 @@ export function ComplexItem({ data, name, opened, edited, jsonpath, info, onSele
...
-										jsonpath={jpath}
-										opened={opened}
-										edited={edited}
-										onSelect={onSelect}
-									/>
-								</li>
-							);
-						})}
-					</ul>
+					{isOpened ? (
```
Comment (ncomont) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js]: 60+ lines for this `return`. Could you split it ?

File: packages/containers/src/ObjectViewer/ObjectViewer.container.js (hunk 1 above)
Comment (jmfrancois) [reply packages/containers/src/ObjectViewer/ObjectViewer.container.js]: no beacuse state is not immutable it's the result of toJS()

Comment (jmfrancois) [reply packages/containers/src/ObjectViewer/ObjectViewer.container.js]: here this is a kind of pristine test

**Pull request merged**